iceoryx_examples/icedelivery/iox_publisher_typed_modern.cpp

```diff
@@ -49,36 +49,44 @@ int main(int argc, char* argv[])
     {
         ++ct;
 
-        // Retrieve a typed sample from shared memory.
-        // Sample can be held until ready to publish.
+        // API Usage #1
+        //  * Retrieve a typed sample from shared memory.
+        //  * Sample can be held until ready to publish.
+        //
         auto result = typedPublisher.loan();
         if (!result.has_error())
         {
             auto& sample = result.get_value();
-            sample->x = ct * 1.1;
-            sample->y = ct * 1.1;
-            sample->z = ct * 1.1;
+            sample->x = ct;
```

Comment:
could you do a placement new, like with the lambda example, just to keep the differences at a minimun

Reply:
The idea was to present different alternatives that work.

Reply:
I just assumed the idea was to have different alternatives how to get the sample, not how to initialize it. By directly assigning instead of doing a placement new, it's harder to tell the actual differences in how the different styles are supposed to be used.

Reply:
@budrus Thoughts ?

Reply:
I will leave it as is for now.

Reply:
@budrus just to clarify, so when I don't have a default c'tor for whatever reason, I cannot use the typed API?